src/atem.ts

```diff
@@ -526,6 +526,52 @@ export class Atem extends EventEmitter {
 		return this.sendCommand(command)
 	}
 
+	listVisibleInputs (mode: 'program' | 'preview', me = 0): number[] {
+		const inputs = new Set<number>()
+
+		// Start with the basics: the surface level of what is in the target ME.
+		this._calcActiveMeInputs(mode, me).forEach(i => inputs.add(i))
+
+		// Loop over the active input IDs we've found so far,
+		// and check if any of them are SuperSources or other nested MEs.
+		// If so, iterate through them and find out what they are showing.
+		// Keep looping until we stop discovering new things.
+		// TODO(Lange): this can almost certainly be cleaner and avoid some duplicate work.
+		let lastSize: number
+		do {
+			lastSize = inputs.size
+			inputs.forEach(inputId => {
+				if (!this.state.inputs[inputId]) {
+					// Data isn't hydrated yet, we'll get 'em next time.
+					return
+				}
+				const portType = this.state.inputs[inputId].internalPortType
+				switch (portType) {
+					case Enums.InternalPortType.SuperSource:
+						const ssrcId = 0 // TODO: don't hardcode this
```

Comment:
Need to figure out an algorithm here like the one being used in the `case Enums.InternalPortType.MEOutput` block below. To figure out this algorithm, I need to know what the input ID of the second SuperSource on the new big boy ATEM is.

Reply:
I believe the ids are 6000 and 6001